ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/TwoPhaseCommitThreadPool.java

```diff
@@ -6,12 +6,35 @@
 
 import com.arjuna.ats.arjuna.common.arjPropertyManager;
 
-import java.util.concurrent.*;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletionService;
+import java.util.concurrent.ExecutorCompletionService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.function.Consumer;
+import java.util.function.Function;
 
+// Remark: this class should be package private - change it or use a different class in BasicAction
 public class TwoPhaseCommitThreadPool {
-    private static final int poolSize = arjPropertyManager.getCoordinatorEnvironmentBean().
-            getMaxTwoPhaseCommitThreads();
-    private static final ExecutorService executor = Executors.newFixedThreadPool(poolSize);
+    private static final ExecutorService executor = initializeExecutor();
+
+    private static ExecutorService initializeExecutor() {
+        return Executors.newFixedThreadPool(
+                arjPropertyManager.getCoordinatorEnvironmentBean().getMaxTwoPhaseCommitThreads());
+    }
+
+    public static boolean isUsingVirtualThreads() {
```

Comment:
Are you definitely OK with this and restartExecutor being in the public API? They look like they are used by the test but maybe you anticipate a use case for business logic to need it?